src/main/scala/com/starrocks/connector/spark/sql/StarrocksRelation.scala

```diff
@@ -83,7 +83,11 @@ private[sql] class StarrocksRelation(
     }
 
     if (filters != null && filters.length > 0) {
-      paramWithScan += (ConfigurationOptions.STARROCKS_FILTER_QUERY -> filterWhereClause)
+      val userFilters = paramWithScan.get(ConfigurationOptions.STARROCKS_FILTER_QUERY)
```

Comment:
1. what if the user configures the filter, but the framework's `filters` is null?
2. add a test to verify it. please refer to `ReadWriteITTest#testSql` for an example.

Reply:
1. use old config, ConfigurationOptions.STARROCKS_FILTER_QUERY
2. ok